Download of activity reports

jasalisbury opened the pull request:
## Description of change

 * Activity reports can now be downloaded in a bulk, instead of 10 at a time
 * Replaced Buttons on the filter to look like actual buttons
 * Tag shows the number of selected reports
 * Reports menu button allows downloading of all reports or just selected reports
 * Add ability to download all `MyAlerts`

## How to test

1) Pull down
2) Approve some reports (over 10)
3) Filter and download approved reports. Verify you are able to download over 10 reports and can download the reports you set a filter for
4) Match the UI against the [design](https://preview.uxpin.com/ad995fd722bd26c0f9a1c8a115f1b41833637308#/pages/138463475)

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-103

## Checklists

### Every PR

<!-- Add details to each completed item -->
- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [x] API Documentation updated
- ~[ ] Boundary diagram updated~
- ~[ ] Logical Data Model updated~
- ~[ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions~

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

dcloud reviewed (commented):
Overall, this looks good, with a few caveats. I think the API endpoint looks solid, the filter and reports menus are more noticeable than the ellipsis menu. Also, thanks for the thorough work on the API documentation!

I added a filter for "topic contains transportation" and it correctly filtered down to the matching 4 reports (of 11 total). Next I removed that filter (clicking the x icon to the left of the filter), but still only 4 reports were showing. It looks like filter removal isn't causing the list of reports to update. **Can we fix this issue?**

Also, I think the reports and filters are a bit confusing, since you can open both and they layer on top of each other. As a user, I expect a menu to close if I click outside of the menu area (I just did that with this Github review menu). I think we'll want a follow-up PR to make these menus better.

I also think you can remove the (hard-to-find) ellipsis menu on the Activity Report table header. It houses a "Download Selected Reports" menu item that is now superseded by the "Export Selected Reports" in the Reports menu.

Also just noticing the small differences in font, colors and padding/sizing of the report/filter buttons from the design. The "X selected" is rendering in a serif font rather than the sans-serif used on the other UI elements. I see a light gray background on the button, but maybe that didn't fit color contrast a11y requirements? The padding/sizing of the buttons and the "X selected" seems cramped. Can that be made bigger? A follow-up PR could tackle this stuff.

So that's one fix for the deselect filters issue, and some follow up PRs/tickets to address some visual and usability issues.

jasalisbury commented on frontend/src/fetchers/index.js:
```diff
@@ -7,12 +7,13 @@ export class HTTPError extends Error {
 }
 
-export const get = async (url) => {
+export const get = async (url, headers = {}) => {
```
It isn't being used. I changed it while trying to use the `activity-reports` endpoint we already have and forgot to change it back once I decided to create separate endpoints.

dcloud reviewed (commented):
I should have caught this earlier, sorry(!), but some of the fields ('reason', 'virtualDeliveryType', etc.) in the export are empty when (I think) they should not be. I think this is due to the `activityReports` service not including those fields. You can see the difference if you compare the output of "Export Table Data…" and "Export Selected Reports…"

jasalisbury commented:
> I should have caught this earlier, sorry(!), but some of the fields ('reason', 'virtualDeliveryType', etc.) in the export are empty when (I think) they should not be. I think this is due to the `activityReports` service not including those fields. You can see the difference if you compare the output of "Export Table Data…" and "Export Selected Reports…"

Should be fixed

jasalisbury commented:
> Overall, this looks good, with a few caveats. I think the API endpoint looks solid, the filter and reports menus are more noticeable than the ellipsis menu. Also, thanks for the thorough work on the API documentation!
> 
> I added a filter for "topic contains transportation" and it correctly filtered down to the matching 4 reports (of 11 total). Next I removed that filter (clicking the x icon to the left of the filter), but still only 4 reports were showing. It looks like filter removal isn't causing the list of reports to update. **Can we fix this issue?**
> 
> Also, I think the reports and filters are a bit confusing, since you can open both and they layer on top of each other. As a user, I expect a menu to close if I click outside of the menu area (I just did that with this Github review menu). I think we'll want a follow-up PR to make these menus better.
> 
> I also think you can remove the (hard-to-find) ellipsis menu on the Activity Report table header. It houses a "Download Selected Reports" menu item that is now superseded by the "Export Selected Reports" in the Reports menu.
> 
> Also just noticing the small differences in font, colors and padding/sizing of the report/filter buttons from the design. The "X selected" is rendering in a serif font rather than the sans-serif used on the other UI elements. I see a light gray background on the button, but maybe that didn't fit color contrast a11y requirements? The padding/sizing of the buttons and the "X selected" seems cramped. Can that be made bigger? A follow-up PR could tackle this stuff.
> 
> So that's one fix for the deselect filters issue, and some follow up PRs/tickets to address some visual and usability issues.

See https://ocio-jira.acf.hhs.gov/browse/TTAHUB-146 & https://ocio-jira.acf.hhs.gov/browse/TTAHUB-147

dcloud reviewed (approved):
:shipit: !

rahearn commented on docs/openapi/paths/index.yaml:
```diff
@@ -20,8 +20,12 @@
   $ref: './updateGrant.yaml'
 '/activity-reports':
   $ref: './activity-reports/activity-reports.yaml'
+'/activity-reports/downloadAll':
```
Looks like this already got updated in the code, but not the docs. Can you just clean this up to match?

rahearn reviewed (changes requested):
Awesome stuff, please just make that one minor docs update.

rahearn reviewed (approved):
🚢
